networkit/cpp/io/NetworkitBinaryWriter.cpp

```diff
@@ -88,33 +168,85 @@ void NetworkitBinaryWriter::write(const Graph &G, const std::string& path) {
 	// Compute encoded size of arrays and store in vector.
 	uint64_t adjSize = 0;
 	uint64_t transpSize = 0;
+	uint64_t adjWeightSize = 0;
+	uint64_t transpWeightSize = 0;
 	std::vector<uint64_t> nrOutNbrs;
 	std::vector<uint64_t> nrInNbrs;
 	std::vector<size_t> adjOffsets;	//Prefix sum of size encoded adj arrays
 	std::vector<size_t> transpOffsets;	//Prefix sum of encoded transposed adj arrays
+	std::vector<size_t> adjWghtOffsets;	//Prefix sum of size encoded adj weights
+	std::vector<size_t> transpWghtOffsets;	//Prefix sum of encoded transposed weights
+
+	auto computeAdjWeightsOffsets = [&] (edgeweight w) {
+		uint8_t tmp [10];
+		switch(weightFormat) {
+			case nkbg::WEIGHT_FORMAT::VARINT:
+				adjWeightSize += encode(w,tmp);
+				break;
+			case nkbg::WEIGHT_FORMAT::DOUBLE:
+				adjWeightSize += sizeof(double);
+				break;
+			case nkbg::WEIGHT_FORMAT::SIGNED_VARINT:
+				{
+					uint64_t weight = encodeZigzag(w);
```

Comment:
Double indentation is not necessary here. Just put the `{` brace on the same line as the `case` and indent this case in the same way as the `case`s without braces.